storybook/src/elements/nuxeo-document-comments/nuxeo-document-comments.stories.js

```diff
@@ -0,0 +1,211 @@
+      'et ultrices posuere cubilia Curae; In ac dui quis mi consectetuer lacinia. Nam pretium turpis et arcu. ' +
+      'Duis arcu tortor, suscipit eget, imperdiet nec, imperdiet iaculis, ipsum. Sed aliquam ultrices mauris. ' +
+      'Integer ante arcu, accumsan a, consectetuer eget, posuere ut,',
+  },
+  {
+    'entity-type': 'comment',
+    parentId: 'doc-id',
+    id: uuid(),
+    numberOfReplies: 0,
+    author: 'Administrator',
+    creationDate: randomDate(new Date(2019, 0, 1), new Date()),
+    text: 'Everything seems fine.',
+  },
+  {
+    'entity-type': 'comment',
+    parentId: 'doc-id',
+    id: uuid(),
+    numberOfReplies: 2,
+    lastReplyDate: new Date(),
+    author: 'Cristiano Ronaldo',
+    creationDate: randomDate(new Date(2019, 0, 1), new Date()),
+    text: 'Best! Top!',
+  },
+  {
+    'entity-type': 'comment',
+    parentId: 'doc-id',
+    id: uuid(),
+    numberOfReplies: 0,
+    author: 'Arnold Schwarzenegger',
+    creationDate: randomDate(new Date(2019, 0, 1), new Date()),
+    modificationDate: undefined,
+    text: 'Strong!',
+  },
+  {
+    'entity-type': 'comment',
+    parentId: 'doc-id',
+    id: uuid(),
+    numberOfReplies: 0,
+    author: 'Cristiano Ronaldo',
+    creationDate: randomDate(new Date(2019, 0, 1), new Date()),
+    text: 'So, lets talk about soccer?',
+  },
+  {
+    'entity-type': 'comment',
+    parentId: 'doc-id',
+    id: uuid(),
+    numberOfReplies: 0,
+    author: 'Mary Poppins',
+    creationDate: randomDate(new Date(2019, 0, 1), new Date()),
+    modificationDate: undefined,
+    text: 'What about music? Lets start a thread here.',
+  },
+  {
+    'entity-type': 'comment',
+    parentId: 'doc-id',
+    id: uuid(),
+    numberOfReplies: 0,
+    author: 'Chuck Norris',
+    creationDate: randomDate(new Date(2019, 0, 1), new Date()),
+    text: 'Some of my jokes: "Chuck Norris can kill two stones with one bird."',
+  },
+].sort((a, b) => (a.creationDate > b.creationDate ? -1 : 1));
+
+const server = fakeServer.create();
+server.respondWith('delete', '/api/v1/id/(?<parentId>\\S*)/@comment/(?<id>(\\S*))');
+server.respondWith('get', '/api/v1/id/(?<id>(\\S*))/@comment/', ({ queryParams }, { id }) => {
+  const comment = commentsSample.find((c) => c.id === id);
+  if (comment) {
+    const replies = [];
+    // To avoid showing threads with more than two levels
+    const repliesSample = commentsSample.filter((c) => c.numberOfReplies === 0);
+    for (let i = 0; i < comment.numberOfReplies; i++) {
+      replies.push(repliesSample[Math.floor(Math.random() * repliesSample.length)]);
+    }
+    return {
+      entries: replies,
+      totalSize: comment ? comment.numberOfReplies : commentsSample.length,
+    };
+  }
+  return {
+    entries: commentsSample.slice(0, queryParams.pageSize === 0 ? commentsSample.length : 10),
+    totalSize: commentsSample.length,
+  };
+});
+server.respondWith('post', '/api/v1/id/(?<id>(\\S*))/@comment/', ({ body }) => {
+  return {
```

Comment:
shouldn't we update the list of comments ?!

Reply:
There is no need, in fact.
We can improve this logic, but I'm not sure if it worths to "complexify" it...